docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,228 @@ The remaining points can happen any time after 1) is complete although note that
+  directory.
+  * This will exclude any `.git`, `.svn`, `log`, share` or `work` directories.
+    Everything else will be copied.
+  * The installation should fail if neither `suite.rc` nor `cylc.flow` exist.
+  * The installation should fail if the target directory is not valid.
+    e.g. you cannot install into `~/cylc-run/my-flow/runN` if `~/cylc-run/my-flow`
+    already contains an installed suite.
+* Version control information will be recorded where relevant (svn/git).
+* The `log, `share`, share/cycle` and `work` directories  will be created
+  following whatever symlink rules are defined for localhost
+  (over-ridable via command line options).
+  * This will continue to support "symlinked alternate run directories"
+    (see [cylc-flow #2935](https://github.com/cylc/cylc-flow/pull/2935)).
+
+`cylc play` - new command to run an installed workflow
+(replacing `cylc run|start`).
+* Will use any relevant options specified as part of the install unless
+  overridden on the command line.
+* Options will look very similar to current `cylc run` command but with no
+  `[START_POINT]` argument (already supported as an option).
+* Fail if the workflow is already running or has completed
+  (need to be able to detect completed run).
+* Continue running the workflow if it was stopped before completion
+  (replaces `cylc restart`).
+* If, for some reason, you want to re-run a workflow from the beginning you will
+  be able to use `cylc clean` to clean out the log directory (and whatever other
+  directories you want removing) after which `cylc play` can be used.
+* Note that `cylc play` will always load the latest workflow definition found in
+  the run directory and will respect any command line options (which may alter
+  the workflow definition). Therefore, continuing a workflow using `cylc play`
+  effectively implies a reload.
+  * The processed `cylc.flow` files will be kept as at present so there will be
+    a record of any changes.
+* Support Rose optional configuration via the environment variable
+  `ROSE_SUITE_OPT_CONF_KEYS` or the option `--rose-opt-conf-key=KEY`.
+
+`cylc reinstall` - new command to re-install workflows.
+* Very similar to the `cylc install` command except that this applies to a
+  previously installed workflow (i.e. you run this command in the directory of
+  the installed workflow, not in the source directory).
+* The rsync will use `--delete` to ensure that any previously installed files
+  which have been removed from the source directory also get removed from the
+  installed workflow.
+  * Note that this will have the effect of removing any files installed via
+    `rose-suite.conf`. They will get recreated by a subsequent reload or play but
+    could affect a running workflow.
+* Any options specified as part of the original install will be retained unless
+  overridden on the command line.
+* Consider a `--dry-run` option to report what would be changed?
```

Comment:
Can we get this script to store a record of changes somewhere so that when people mess their workflows up we can see what they've done?